plugins/Marketplace/vue/src/types.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+export type TObject = Record<string, unknown> | Record<string, never>;
+export type TObjectArray = TObject[] | [] | null;
+export type TNumberOrString = string | number | null;
+
+export interface IPluginShopVariation {
```

Comment:
Yes, there's no precedent yet, but I think it's a good practice and it allows everyone to see at first glance what kind of definition it is compared to a name that doesn't have that indication. Happy to be overruled on this if the team feels strongly against that.

Reply:
You can also see that I kept the interface defined for the component without the prefix for consistency with other components and only used the `I` and `T` prefixes for partials defining the main component interface.

Reply:
I would only say we should align on the direction and have that decision enforced ([eslint](https://github.com/typescript-eslint/typescript-eslint/blob/v7.7.1/packages/eslint-plugin/docs/rules/naming-convention.mdx#enforce-that-type-parameters-generics-are-prefixed-with-t)) as much as possible.

Otherwise it may get a wild mix of everything and there are no real arguments to have one side chosen over the other one except personal preference.

Do @sgiehl / @caddoo have an opinion on that one?

Reply:
If this was the only thing blocking the merge we can merge it as is and come back to it later should we chose to change the approach.

Reply:
Not blocking anything. Will create an internal ticket so we can eventually align on the direction to take.